core/node/rpc/server.go

```diff
@@ -581,6 +589,49 @@ func (s *Service) initCacheAndSync() error {
 	return nil
 }
 
+func (s *Service) initScrubbing(ctx context.Context) (err error) {
+	log := dlog.FromCtx(ctx).With("Func", "initScrubbing")
+	ctx, cancel := context.WithCancel(ctx)
+	s.onClose(cancel)
+
+	scrubEventQueue := make(chan *rules.DerivedEvent, 100)
+	s.scrubEventQueue = scrubEventQueue
+	s.scrubTaskProcessor, err = scrub.NewStreamScrubTasksProcessor(
+		ctx,
+		s.cache,
+		scrubEventQueue,
+		s.chainAuth,
+		s.config,
+	)
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case event := <-s.scrubEventQueue:
+				err := s.addEventPayload(ctx, event.StreamId, event.Payload)
```

Comment:
It seems it's possible to remove this channel:
rename addEventPayload to AddEventPayload.
instead of scrubEventQueue pass s.
On the scrub side define something like 
```go
type EventAdder interface {
  AddEventPayload
}
```
and use it instead of channel.